src/js/stores/AppFormTransforms.js

```diff
@@ -1,9 +1,32 @@
+var Util = require("../helpers/Util");
+
+function trimDuplicableRows(rows) {
+  var trimmedRows = rows.map((row) => {
+    return Object.keys(row).reduce((memo, key) => {
+      if (row[key] != null
+        && key !== "consecutiveKey"
+        && !Util.isEmptyString(row[key])) {
+        memo[key] = row[key];
+      }
+      return memo;
+    }, {});
+  }).filter((row) => Object.keys(row).length);
+
+  if (trimmedRows.length) {
+    return trimmedRows;
+  }
+  return null;
```

Comment:
This must return an empty array, if the user deleted the last row else the app definition won't be overriden on the API.